### PR TITLE
tests: Assert some assumptions about the default role behaviour

### PR DESCRIPTION
**Pull request by martinpitt**
Check the socket state, that cockpit's web server works, and that the
role does not create a config file.

-----
To test this locally, I started a standard Fedora 34 cloud image VM, which my ssh config knows as `c`, created a simple inventory:
```
❱❱❱ cat /tmp/i 
[all]
c
```
and then ran

    ansible-playbook -i /tmp/i tests_default.yml

in the tests/ directory. This works fine. Now I would like to see how that looks in your CI, as these integration tests don't run by default (as in #33), or there are a lot of failures (as in #32, although that looked infra related).

Also, is this general shape for the tests okay? I am planning to add some more, like a tests/test_config.yml for the config writer. Then I'd add a feature to directly specify a banner test (which will create a banner file and point `cockpit_config.Session.Banner` to that), as discussed in our recent meeting.

After that I feel ready to address the bigger point of adding certificate support. But this really should be in CI properly first -- right now there are zero assertions beyond "the role does not fail".

### CODE REVIEW

**Comment by martinpitt** [comment]
Ignored the [ansiblelint errors](https://github.com/linux-system-roles/cockpit/pull/34/checks?check_run_id=3857099384#step:5:81), there is no better way to do these checks.

File: tests/tests_default.yml [hunk 1]
```diff
@@ -2,3 +2,28 @@
 - hosts: all
   roles:
     - linux-system-roles.cockpit
+
+  tasks:
+    - name: "test: socket is active"
+      command: systemctl is-active cockpit.socket  # noqa 301 303 command-instead-of-module no-changed-when
+      register: result
+      failed_when: result.failed
```
Comment (richm) [inline tests/tests_default.yml:10]: I think you can remove the `register` and `failed_when` here.  You can get rid of the `noqa no-changed-when` by adding `changed_when: false`
If you don't use a `:` in the value of the `name:`, you don't need to use double quotes e.g.
`name: test - socket is active`

Comment (martinpitt) [reply tests/tests_default.yml]: Indeed, thanks! I cleaned this up as you suggested, much better.

**Comment by martinpitt** [comment]
I added the test cleanup that @richm asked about in https://github.com/linux-system-roles/cockpit/pull/35#discussion_r726202547 . Tested in a local F34 VM

**Pull request merged**